io/net/infiniband/mofed_install_test.py

```diff
@@ -72,10 +77,10 @@ def setUp(self):
                     detected_distro.name, detected_distro.version, detected_distro.release)
                 patterns = [host_distro_pattern]
                 for pattern in patterns:
-                    temp_string = process.getoutput(
-                        "curl --silent %s" % (self.iso_location), verbose=False, shell=True, ignore_status=True)
+                    response = urllib.request.urlopen(self.iso_location)
```

Comment:
can we try to see if `asset_fetch` module from avocado work?

Reply:
We are matching the MOFED ISO to be installed based on the distro OS already installed in the server. 
So, for this scenario we are getting the ISO from the http or https repository wherein many different ISO's will be available for different distro releases.

We are building the full iso path once we match the OS level with the corresponding MOFED ISO and then download the same to the server to install the same.

We are using fetch_asset() function of avocado after we get the full iso path and pass the same to it as can be seen in lines 87 and line 53 of this script.

Reply:
@PraveenPenguin `self.iso = self.fetch_asset(self.iso, expire='10d')` its being used already... but before that this code to handle https path